src/Neo/SmartContract/Native/CryptoLib.cs

```diff
@@ -33,6 +33,34 @@ public sealed partial class CryptoLib : NativeContract
 
         internal CryptoLib() : base() { }
 
+        /// <summary>
+        /// Recovers the public key from a secp256k1 signature in a single byte array format.
+        /// </summary>
+        /// <param name="message">The original message that was signed.</param>
+        /// <param name="signature">The 65-byte signature in format: r[32] + s[32] + v[1]. 64-bytes for eip-2098, where v must be 27 or 28.</param>
+        /// <param name="hashAlgorithm">The hash algorithm to be used (SHA256 or Keccak256).</param>
+        /// <returns>The recovered public key in compressed format, or null if recovery fails.</returns>
+        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 10, Name = "recoverSecp256K1")]
+        public static byte[] RecoverSecp256K1(byte[] message, byte[] signature, HashAlgorithm hashAlgorithm)
```

Comment:
Based on the answer [here](https://github.com/neo-project/neo/pull/3633#issuecomment-2609797233), I thought you might use the function signature where the message is already hashed (i.e., `(byte[] hash, byte[] signature)`). Do you have a preference for this function signature then?

Reply:
`HashAlgorithm.None` ?

Reply:
> `HashAlgorithm.None` ?

That's one way of supporting recovery with an already hashed message, yes. I'm fine with that. 👍🏼